src/managers/SubscriptionManager.py

```diff
@@ -49,3 +49,179 @@ def update_subscription(self, subscription_id, subscription_update_dto):
+            logging.error(f"Database error getting active subscriptions: {e}")
+            raise
+        except ProgrammingError as e:
+            logging.error(f"Invalid SQL query: {e}")
+            raise
+        except Exception as e:
+            logging.error(f"Error getting active subscriptions: {e}")
+            raise
+
+    def get_pending_subscriptions(self):
+        try:
+            return self.subscription_store.get_pending_subscriptions()
+        except OperationalError as e:
+            logging.error(f"Database error getting pending subscriptions: {e}")
+            raise
+        except ProgrammingError as e:
+            logging.error(f"Invalid SQL query: {e}")
+            raise
+        except Exception as e:
+            logging.error(f"Error getting pending subscriptions: {e}")
+            raise
+
+    def get_expired_subscriptions(self):
+        try:
+            return self.subscription_store.get_expired_subscriptions()
+        except OperationalError as e:
+            logging.error(f"Database error getting expired subscriptions: {e}")
+            raise
+        except ProgrammingError as e:
+            logging.error(f"Invalid SQL query: {e}")
+            raise
+        except Exception as e:
+            logging.error(f"Error getting expired subscriptions: {e}")
+            raise
+
+    def get_cancelled_subscriptions(self):
+        try:
+            return self.subscription_store.get_cancelled_subscriptions()
+        except OperationalError as e:
+            logging.error(f"Database error getting cancelled subscriptions: {e}")
+            raise
+        except ProgrammingError as e:
+            logging.error(f"Invalid SQL query: {e}")
+            raise
+        except Exception as e:
+            logging.error(f"Error getting cancelled subscriptions: {e}")
+            raise
+
+    def get_subscription_by_user_id(self, user_id):
+        try:
+            return self.subscription_store.get_subscription_by_user_id(user_id)
+        except OperationalError as e:
+            logging.error(f"Database error getting subscription by user ID: {e}")
+            raise
+        except ProgrammingError as e:
+            logging.error(f"Invalid SQL query: {e}")
+            raise
+        except Exception as e:
+            logging.error(f"Error getting subscription by user ID: {e}")
+            raise
+
+    def get_subscription_by_product_id(self, product_id):
+        try:
+            return self.subscription_store.get_subscription_by_product_id(product_id)
+        except OperationalError as e:
+            logging.error(f"Database error getting subscription by product ID: {e}")
+            raise
+        except ProgrammingError as e:
+            logging.error(f"Invalid SQL query: {e}")
+            raise
+        except Exception as e:
+            logging.error(f"Error getting subscription by product ID: {e}")
+            raise
+
+    def get_subscription_by_status(self, status):
```

Comment:
Remove the other methods and leave only this one

Reply:
check this with Joro (should i delete all other methods or those from the last commit)